### ISSUE TITLE
Projects generated with openapi generator do not build

### ISSUE THREAD

**Issue by romain-grecourt** [oracle/helidon#4110]
When generating a project using using OpenAPI [generator](https://openapi-generator.tech/docs/generators/jaxrs-spec/), the generated project does not build.

The generated code uses Jackson but the generated pom does depend on Jackson or the Jersey Jackson media provider.

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /private/tmp/my-project/src/main/java/org/openapitools/model/GreetResponse.java:[3,21] package org.joda.time does not exist
[ERROR] /private/tmp/my-project/src/main/java/org/openapitools/model/GreetResponse.java:[8,40] package com.fasterxml.jackson.annotation does not exist
[ERROR] /private/tmp/my-project/src/main/java/org/openapitools/model/GreetResponse.java:[9,40] package com.fasterxml.jackson.annotation does not exist
[ERROR] /private/tmp/my-project/src/main/java/org/openapitools/model/GreetResponse.java:[10,40] package com.fasterxml.jackson.annotation does not exist
[ERROR] /private/tmp/my-project/src/main/java/org/openapitools/model/GreetResponse.java:[11,40] package com.fasterxml.jackson.annotation does not exist
```

### Steps

Install `openapi-generator-cli`:
```bash
mkdir -p ~/bin/openapitools
curl https://raw.githubusercontent.com/OpenAPITools/openapi-generator/master/bin/utils/openapi-generator-cli.sh > ~/bin/openapitools/openapi-generator-cli
chmod u+x ~/bin/openapitools/openapi-generator-cli
export PATH=$PATH:~/bin/openapitools/
```

Create `greeting.yaml`:
```yaml
info:
  title: Hello World Greet Helidon MP Application
  version: '1.0'
openapi: 3.0.3
paths:
  /greet:
    get:
      summary: Gets default greet message
      operationId: getDefaultMessage
      tags:
        - greet
      responses:
        '200':
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/GreetResponse'
          description: OK
  /greet/{name}:
    get:
      summary: Gets message based on the parameter name
      operationId: getMessage
      tags:
        - greet
      parameters:
        -
          in: path
          name: name
          required: true
          schema:
            type: string
      responses:
        '200':
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/GreetResponse'
          description: OK
components:
  schemas:
    GreetResponse:
      properties:
        message:
          type: string
        date:
          format: date
          type: string
      type: object
```

Generate the project:
```bash
openapi-generator-cli generate \
  -g jaxrs-spec \
  --library helidon \
  -p sourceFolder=src/main/java \
  -o my-project  \
  -i greeting.yaml
```

Build the generated project:
```bash
mvn package
```

**Comment by tjquinno** [comment]
Is this a duplicate of this issue? https://github.com/OpenAPITools/openapi-generator/issues/7095

**Comment by romain-grecourt** [comment]
That would have been a workaround, if it worked :D
Either way we need to make sure we setup Jackson in the pom correctly when used.

**Comment by spericas** [comment]
I have found other problems besides the Jackson issue. Summary:

1. Generator creates an empty `Application` subclass with a path specified in `@ApplicationPath`. This class is currently ignored by Helidon because it does not have a BDA such as `@ApplicationScoped`. 
2. A consequence of (1) is that the application path is ignored, so all resources are in a different location
3. The sources are generated under `src/gen/java` which is not a compiler source root for Maven by default (pom can be updated).
4. Generation of `@Valid` should be turned off by default
5. Indentation, especially in model classes, is ugly and inconsistent

**Comment by tvallin** [comment]
Hi @spericas,
Do you mind if I open a new epic issue containing the issues you mentioned ?
So I can make a proper PR with the fixes.

**Comment by tjquinno** [comment]
A quick note...

> 3. The sources are generated under src/gen/java which is not a compiler source root for Maven by default (pom can be updated).

This is the default when the generator is run from the command line. It can be overridden by a command-line setting.

 The Maven plug-in for the generator automatically sets the generated code location to a more proper directory under `generated-sources`.

**Comment by spericas** [comment]
@tjquinno IMO we need to provide a nicer out-of-the-box experience for our customers (generator, plugin or our CLI); having to specify multiple command-line settings is cumbersome and error prone (things like `--additional-properties="interfaceOnly=true"` for example). Ideally, we should run a simple command and have a runnable Helidon application without any issues. Also, I think the generated code should include unit tests.

**Comment by tjquinno** [comment]
@spericas Agreed on all counts.

If we have something that drives the generator for the developer, then that driver could take care of all the settings specific to Helidon.

Even so, apart from such a driver, we want the direct use of the generator to be easy for Helidon devs.

The framework-specific Java code in the generator has lots of flexibility in controlling the generated code. Specifically, that code can override the defaults for option settings and control which templates are used during generation and, for each template, where the generated file goes.

That means, for example, that a Helidon-_specific_ library can make the generation happen exactly the way we want. If we create a _new_ Helidon-_friendly_ library--such as, for example, for clients a `java/microprofile-jersey` library--then we can similarly control the generation settings so they are what we want.

If we want to modify an existing library to be more Helidon-friendly, then of course we cannot just change its defaults to suit our purposes. We could do something like adding an additional property such as, say, `helidon`, the presence of which would turn on the Helidon-friendly behavior. (In fact we can think about whether that would be a good idea as well for libraries we modify _or_ create.)

As for adding tests... The Java code for a specific generator influences/controls what files are generated and where. For example, here is an example of that in some of the source I referenced in the internal gist I circulated earlier: https://github.com/OpenAPITools/openapi-generator/blob/498ba58881e29c2c18ac5a3233f683a62c077f90/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaJAXRSSpecServerCodegen.java#L216-L223)

Adding unit test generation requires adding the unit test file templates and updating  code similar to what I linked to above in the generator's Java code to add those templates to the generation. The link above show files generated into directories under `src/main/...` but of course the generator code for adding unit tests could easily specify `src/test...` instead.
